File: test/pages/HomePage.java
package pages;

import static org.fest.assertions.Assertions.*;
import org.fluentlenium.core.FluentPage;
import org.openqa.selenium.WebDriver;
import static org.fluentlenium.core.filter.FilterConstructor.*;

public class HomePage extends FluentPage {

    public HomePage(WebDriver webDriver, int port, String baseUrl) {
        super(webDriver);
        withDefaultUrl(baseUrl);
    }

    @Override
    public String getUrl() { return ""; }

    @Override
    public void isAt() {
        assertThat(title()).isEqualTo("Daily Soccer");
    }

    public void clickOnLogin() {
        find(".navbar-right").find("li a", withText("Login")).click();
    }

    public void clickOnSignUp() {
        find(".navbar-right").find("li a", withText("Join")).click();
    }
}

File: test/pages/HomePage.scala
package pages

class HomePage extends SharedPage {
  val TITLE   = "Daily Soccer"
  val LOGIN_TEXT  = "Login"
  val SIGNUP_TEXT = "Join"

  val url = SharedPage.baseUrl

  def open = {
    go to url
    this
  }

  def isAt = {
    pageTitle should be (TITLE)
    this
  }

  def clickOnLogin = {
    click on linkText(LOGIN_TEXT)
    new LoginPage().isAt
  }

  def clickOnSignUp = {
    click on linkText(SIGNUP_TEXT)
    new SignUpPage().isAt
  }
}
